api/src/controllers/search.js

```diff
@@ -738,6 +738,17 @@ const generateMatchesForAggregation = async function (and, or, nor, searchProper
 };
 
 const executeQuery = async function (args, res, next) {
+    // It's possible to bring down the API if you pass in an invalid item ID, or null.
+    // a post to "api/search?dataset=Item&_id=some-non-object-id" will cause the server to die.
+    // This will still let any 12 char value through (it passes isValid)
```

Comment:
To protect against this, the final call should be wrapped in a try/catch and handled gracefully, instead of allowing the aggregation to crash the API.

Reply:
Ahh I was mistaken here, a 12 char value will not crash the api.  It will return no results. I will implement the try catch, to prevent other errors crashing it out.